web/api/v1/api.go

```diff
@@ -2099,3 +2114,22 @@ func toHintLimit(limit int) int {
 	}
 	return limit
 }
+
+func truncateResults(result *promql.Result, limit int) *promql.Result {
+	switch v := result.Value.(type) {
+	case promql.Matrix:
+		if len(v) > limit {
+			result.Value = v[:limit]
+		}
+	case promql.Vector:
+		if len(v) > limit {
+			result.Value = v[:limit]
+		}
+	default:
+		// No truncation for other types (Scalars or Strings)
```

Comment:
End sentence with a period.

Reply:
In different news, we don't really need this default section. We return the result below anyway.  I suggest to remove the `default` branch and have a comment below that the returned result is unchanged for other types.

Reply:
Should we write tests specifically for `truncateResults`, or would it be better to focus on `query` and `queryRange` where `truncateResults` is indirectly tested?

Reply:
I would think a test on the query level (like in #13396) should suffice.

Reply:
I have implemented the following things according to the changes requested:
- updated the documentation
- added a warning if the result gets truncated

and I'm working on the test case now.

Reply:
In the testEndpoints function, there are no test cases for matrix/vector responses returning multiple values that I can use to test the limit parameter. I added the limit expecting it to constrain FPoint values, but it didn't work as intended. Any ideas for predicting or verifying the response?

https://github.com/prometheus/prometheus/blob/main/web/api/v1/api_test.go#L1210-L1231

Is there a reason why we are not covering testing matrix/vector for Query API and queryRange API? Is it because that's not possible in the test suite?